### ISSUE TITLE
appset generator logic from configmap

### ISSUE THREAD

**Issue by waheedshahani** [argoproj-labs/applicationset#437]
### Feature Request
**TL;DR**
Allowing generators to read logic from configmap specially elements of list generator.

**TS;DR**
An application set needs generators as well as template. This full manifest of appset could become more complicated and could easily become few hundered lines when using apps of apps pattern , helm with large list of values etc.... Putting all the logic in one manifest file can result in touching this manifest even if there are changes needed in generators logic only. This could becomes problematic in big organization where you have separate operations team with less skills.

In order to avoid confusion/complexity and to separate generator logic definition in configmap would be useful. 

In addition to above use case, we could have more granular RBAC on who can do changes in configmap v/s who can do changes in appset. Allowing everyone (even automated systems) to update appset logic seems too much dangerous.
